setup.py

```diff
@@ -35,7 +35,7 @@
     "markupsafe==2.0.1",
     "coverage",
     "furo",
-    "mypy==0.991",
+    "mypy==1.11.1",
```

Comment:
We generally only pin to a specific version if an incompatibility arises, otherwise the current major
```suggestion
    "mypy<2",
```